python/dataset_plotting.py

```diff
@@ -0,0 +1,239 @@
+                height=800*ratio,
+                xaxis = dict(
+                    range = [axmin[0],axmax[0]],
+                    title = "{} [{}]".format(axlab[0][0],axlab[0][1])),
+                yaxis = dict(
+                    scaleanchor = "x",
+                    scaleratio = ratio,
+                    range = [axmin[1],axmax[1]],
+                    title = "{} [{}]".format(axlab[1][0],axlab[1][1]))
+            )
+
+        )
+
+        def update_z(zpos):
+            fig.data[0].z = a[:,:,zpos]
+
+        # Add a slider that updates the slice plane
+        slider = interactive(update_z, zpos=(0, nx[2]-1, 1))
+        vb = VBox((fig, slider))
+        vb.layout.align_items = 'center'
+        return vb
+    else:
+        raise RuntimeError("Unsupported number of dimensions in sliceviewer.")
+
+#===============================================================================
+
+# Plot a 1D spectrum
+# Inputs can be:
+# 1. d (=dataset)
+# 2. d, var (=string)
+# 3. [d1, d2]
+# 4. [[d1, var1], [d2, var2]]
+# 5. [[d1, var1], [d2]]
+#
+# TODO: find a more general way of handling arguments to be sent to plotly,
+# probably via a dictionay of arguments
+def plot_1d(input_data, var=None, logx=False, logy=False, logxy=False):
+
+    arr = []
+    # Case of a single dataset
+    if type(input_data) is dataset.Dataset:
+        arr.append([input_data, var])
+    # The more general case: there can be many different types of lists
+    elif type(input_data) is list:
+        # Go through the list items recursively:
+        for item in input_data:
+            if type(item) is dataset.Dataset:
+                arr.append([item, None])
+            elif type(item) is list:
+                ktem = [None, None]
+                for jtem in item:
+                    if type(jtem) is dataset.Dataset:
+                        ktem[0] = jtem
+                    elif type(jtem) is str:
+                        ktem[1] = jtem
+                    else:
+                        raise RuntimeError("Bad data type in input of plot_1d")
+                arr.append(ktem)
+
+    # arr now contains a list of [dataset,var] pairs
+    # We now construct a list of [x,y] pairs
+    # TODO: check that all x coordinates are the same
+    data = []
+    for item in arr:
+        # Scan the datasets
+        coords = []
+        fields = []
+        for var in item[0]:
+            if var.is_coord:
+                coords.append(var)
+            if var.is_data:
+                fields.append(var)
+                if var.name == item[1]:
+                    ifield = len(fields) - 1
+
+        if (item[1] is None) and (len(fields) > 1):
+            raise RuntimeError("More than one data field found! Please specify which one to display")
+
+        y = fields[ifield].numpy
+        ylab = fields[ifield].unit.name
+        name = fields[ifield].name
+
+        # Now find the appropriate coordinate
+        coord_not_found = True
+        for c in coords:
```

Comment:
With what I said above, we should have only a single coordinate to handle? Or is there a case for having more than one?